Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/ClientGameNetPortal.cs

```diff
@@ -239,22 +243,33 @@ public async void StartClientUnityRelayModeAsync(GameNetPortal portal, string jo
                     }
                     catch (Exception e)
                     {
-                        OnUnityRelayJoinFailed?.Invoke(e.Message);
-                        // todo remove the above callback and get the below uncommented when UI is its own assembly
-                        // var menuUI = MainMenuUI.Instance;
-                        // if (menuUI)
-                        // {
-                        //     menuUI.PushConnectionResponsePopup("Unity Relay: Join Failed", $"{e.Message}", true, true);
-                        // }
-                        throw;
+                        if (cancellationToken.IsCancellationRequested)
+                        {
+                            Debug.Log("Unity Relay join failed, but was cancelled: " + e.Message);
+                        }
+                        else
+                        {
+                            OnUnityRelayJoinFailed?.Invoke(e.Message);
+
+                            // todo remove the above callback and get the below uncommented when UI is its own assembly
+                            // var menuUI = MainMenuUI.Instance;
+                            // if (menuUI)
+                            // {
+                            //     menuUI.PushConnectionResponsePopup("Unity Relay: Join Failed", $"{e.Message}", true, true);
+                            // }
+                            throw;
+                        }
                     }
 
                     break;
                 default:
                     throw new Exception($"unhandled relay transport {chosenTransport.GetType()}");
             }
 
-            ConnectClient(portal);
+            if (!cancellationToken.IsCancellationRequested)
+            {
+                ConnectClient(portal);
```

Comment:
Thinking of this, there's no way to cancel the connection itself? Cause that part too could have delays and issues.
If you test with clumsy or even try to connect to an invalid address, there's no way to cancel that once that's launched?

Reply:
From my investigation, I haven't found a way to cancel the connection itself. However, If we cancel the operation after this part has started, we still call NetworkManager's Shutdown method. On the user side, it seems to properly cancel the connection. I just tested it again with direct IP and with UnityRelay and it looks like it works fine

Reply:
ah so "Shutdown" would be what cancels the thing. ok
it might be worth it to add a quick comment on this for users investigating the same thing